assets/src/edit-story/components/form/advancedDropDown/dropDown.js

```diff
@@ -0,0 +1,231 @@
+    throw new Error(
+      'Dropdown initiated with invalid params: options or getOptionsByQuery has to be set'
+    );
+  }
+  // If search is not enabled, always display all options.
+  if (!hasSearch) {
+    primaryOptions = options;
+  }
+  const ref = useRef();
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const closeDropDown = useCallback(() => {
+    setIsOpen(false);
+    // Restore focus
+    if (ref.current) {
+      ref.current.focus();
+    }
+  }, []);
+  const toggleDropDown = useCallback(() => setIsOpen((val) => !val), []);
+  // Must be debounced to account for clicking the select box again
+  // (closing in useFocusOut and then opening again in onClick)
+  const [debouncedCloseDropDown] = useDebouncedCallback(closeDropDown, 100);
+
+  const handleSelect = useCallback(
+    (option) => {
+      onChange(option);
+      setIsOpen(false);
+      ref.current.focus();
+    },
+    [onChange]
+  );
+
+  const handleKeyPress = useCallback(
+    ({ key }) => {
+      if (
+        !isOpen &&
+        key === 'ArrowDown' &&
+        document.activeElement === ref.current
+      ) {
+        setIsOpen(true);
+      }
+    },
+    [isOpen]
+  );
+
+  const selectedOption = primaryOptions.find(({ id }) => id === selectedId);
+  return (
+    <Container onKeyDown={handleKeyPress}>
+      <DropDownSelect
+        onClick={toggleDropDown}
+        aria-pressed={isOpen}
+        aria-haspopup
+        aria-expanded={isOpen}
+        ref={ref}
+        lightMode={lightMode}
+        {...rest}
+      >
+        <DropDownTitle>{selectedOption?.name || placeholder}</DropDownTitle>
+        <DropDownIcon />
+      </DropDownSelect>
+      {!disabled && (
+        <Popup anchor={ref} isOpen={isOpen} fillWidth={DEFAULT_WIDTH}>
+          <OptionsContainer
+            isOpen={isOpen}
+            value={selectedId}
+            onSelect={handleSelect}
+            onClose={debouncedCloseDropDown}
+            getOptionsByQuery={getOptionsByQuery}
+            hasSearch={hasSearch}
+            onObserve={onObserve}
+            options={options}
+            primaryOptions={primaryOptions}
+            primaryLabel={primaryLabel}
+            priorityOptions={priorityOptions}
+            priorityLabel={priorityLabel}
+            renderer={renderer}
+          />
+        </Popup>
+      )}
+    </Container>
+  );
+}
+
+DropDown.propTypes = {
+  selectedId: PropTypes.any,
+  onChange: PropTypes.func.isRequired,
+  lightMode: PropTypes.bool,
+  placeholder: PropTypes.string,
+  disabled: PropTypes.bool,
+  options: PropTypes.array,
+  hasSearch: PropTypes.bool,
+  getOptionsByQuery: PropTypes.func,
+  onObserve: PropTypes.func,
+  primaryOptions: PropTypes.array,
+  primaryLabel: PropTypes.string,
+  priorityOptions: PropTypes.array,
+  priorityLabel: PropTypes.string,
+  renderer: PropTypes.func,
+};
+
+DropDown.defaultProps = {
+  value: '',
+  lightMode: false,
+  placeholder: __('Select an Option', 'web-stories'),
```

Comment:
Can we inline these in the definition instead? `defaultProps` will [be retired at some point anyway](https://twitter.com/dan_abramov/status/1133878326358171650).